### ISSUE TITLE
NeuralNetworksExtensions.h

### ISSUE THREAD

**Issue by mdenna-synaptics** [android/ndk#1503]
Hi,
I would like to use NNAPI extensions in my application.
Unfortunately it looks like the NeuralNetworksExtensions.h header is not present in the NDK (I checked r21 and r22).
Furthermore the corresponding extension symbols (e.g. ANeuralNetworksDevice_getExtensionSupport) are not present in libneuralnetworks.so

Is this expected? What's the reason for this?
What should I do in order to use NNAPI extension functions?

Best regards,
Maurizio

**Comment by DanAlbert** [comment]
Yes, it's expected. Those are vendor APIs, not app APIs (see the comment at the top of that header).

**Comment by ayounes-synaptics** [comment]
Hi Dan,
Thanks for your answer but it would be nice if you could reconsider. The NN extension API is completely self contained and does not depend on any third party API. Right now this forces us to deliver our own copy of libneuralnetworks.so as part of our SDK. Please note that the extensions, if any, are not implemented in the library itself but in one of the NN HAL.; the extensions are thus runtime discovered.
I would understand if you don't want to deliver the extension header but it would be very nice to get the library with the extension symbols included.

**Comment by DanAlbert** [comment]
It's not my decision to make. I just ship them. You'll need to file a platform bug following b.android.com.
